src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -272,9 +275,59 @@ private Long getPollingFrequency(Topic pollingFrequencyTopic) {
 
     @Override
     protected void shutdown() {
+        saveDeploymentQueueToConfig(); // Save any deployments in queue for next startup
         receivedShutdown.set(true);
```

Comment:
receivedShutdown controls the loop in `startup` - should you save after it is set to `true` so that loop isn't trying to process a deployment while you are trying to save the queue?

Reply:
I was actually concerned that setting `receivedShutdown` to `true` would trigger `finishCurrentDeployment()` and set `currentDeploymentTaskMetadata = null` before I had a chance to save it to the config.

Reply:
So no, I don't think so.

The queue saving logic wants `currentDeploymentTaskMetadata.getDeployment()`. The deployment processing loop only modifies `currentDeploymentTaskMetadata` here, in `createNewDeployment()`:

https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/b273fa79d962fc9aedcf0abccbef01d540662b24/src/main/java/com/aws/greengrass/deployment/DeploymentService.java#L502

`createNewDeployment()` only gets invoked when `currentDeploymentTaskMetadata` is `null`, which happens when `finishCurrentDeployment()` is invoked.

Reply:
how would setting it to true trigger the finishCurrentDeployment condition?

Reply:
ah, i misread. `receivedShutdown==true` effectively just stops the loop, which is fine.

i'll swap the order of loading/setting the flag.